app/models/identity.rb

```diff
@@ -2,6 +2,10 @@ class Identity < ActiveRecord::Base
   include NonNullUuid
 
   belongs_to :user
+  belongs_to :sp,
```

Comment:
If that's what we want, then yes, please, redesign the Identity and Service Provider models.